mlir/include/Conversion/LoSPNtoCPU/Vectorization/SLP/Util.h

```diff
@@ -0,0 +1,150 @@
+//==============================================================================
+// This file is part of the SPNC project under the Apache License v2.0 by the
+// Embedded Systems and Applications Group, TU Darmstadt.
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// SPDX-License-Identifier: Apache-2.0
+//==============================================================================
+
+#ifndef SPNC_MLIR_INCLUDE_CONVERSION_LOSPNTOCPU_VECTORIZATION_SLP_UTIL_H
+#define SPNC_MLIR_INCLUDE_CONVERSION_LOSPNTOCPU_VECTORIZATION_SLP_UTIL_H
+
+#include "SLPGraph.h"
+#include "LoSPN/LoSPNOps.h"
+#include "mlir/IR/Operation.h"
+
+namespace mlir {
+  namespace spn {
+    namespace low {
+      namespace slp {
+
+        /// TODO: make GlobalOptions.h available to the SLP vectorization in CMake.
```

Comment:
During development of the SLP options I thought that inclusion of `GlobalOptions.h` would be much better because its .cpp already defines default values for the options declared in the lines following the TODO. The current solution for accessing options [like this](https://github.com/esa-tu-darmstadt/spn-compiler/blob/a98cc6ae05a0a6e37ef9c305a62c57f1aff70a83/mlir/lib/Conversion/LoSPNtoCPU/Vectorization/SLP/SLPGraphBuilder.cpp#L69) requires a) redeclaring every SLP option and b) redefining their default values ([here](https://github.com/esa-tu-darmstadt/spn-compiler/blob/a98cc6ae05a0a6e37ef9c305a62c57f1aff70a83/mlir/lib/Conversion/LoSPNtoCPU/Vectorization/SLP/Util.cpp#L17)).

Reply:
Do you think that the use of the MLIR-infrastructure for pass options (see the general discussion in the PR), would help here? 

Reversing the dependency direction or including `GlobalOptions.h` is not really an option, so we can either use the MLIR pass options or simply keep the redeclaration and remove the TODO here.

Reply:
The new infrastructure is now used.

Reply:
In addition to using the new infrastructure I also removed the global option access from the mentioned `Util.h` file in 5b6da110f75a0fd6f7cf22bbb53ee6e25ea72a87. Every class that uses SLP related options now stores them as private members.